pkg/clusteragent/clusterchecks/stores.go

```diff
@@ -0,0 +1,85 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build clusterchecks
+
+package clusterchecks
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/clusterchecks/types"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// clusterStore holds the state of cluster-check management.
+// Lock is to be held by the dispatcher so it can make atomic
+// operations involving several calls.
+type clusterStore struct {
+	sync.RWMutex
+	digestToConfig map[string]integration.Config // All configurations to dispatch
+	digestToNode   map[string]string             // Node running a config
```

Comment:
This map alone can allow to rebuild the dispatching state (assuming our digests are stable).
We should be able to store it in a file and restore from it